frontend/apps/reader/modules/readerdictionary.lua

```diff
@@ -64,20 +65,90 @@ local function tidy_markup(results)
     return results
 end
 
+function ReaderDictionary:cleanSelection(text, is_single_word)
+    -- Will be used by ReaderWikipedia too (with is_single_word false)
+    -- We do multiple times the same replacements, which is most of the time overkill,
+    -- but sometimes provices better cleaning
```

Comment:
Can i just put there in comments some sample selections crengine has returned ?
Don't ask me to do some unit test please :| I dont think it's that critical to be consistent here. It's a mixture of crengine word detection algorithm (which may be greedy), and real life epub editors' various typography rules (using various kind of dash, unbreakable spaces, or not)